main.go

```diff
@@ -229,12 +235,52 @@ func visit(url *url.URL) {
 				// 30x but no Location to follow, give up.
 				return
 			}
-			log.Fatal("unable to follow redirect: %v", err)
+			log.Fatalf("unable to follow redirect: %v", err)
 		}
 		visit(loc)
 	}
 }
 
+func responseBody(response io.ReadCloser) string {
```

Comment:
Also, even though it's a private function a comment describing what it does never hurts.
```go
// readResponseBody spools the http response body to disk or to screen depending
// on the value of SOME FLAG.
```

When written like this, it's clear that the function violates SRP, it does two things, depending on the value of a global flag. Actually it does two things, because depending on the `showBody` flag, it either writes the data to a temp file, and prints out the name of the temp file, or it just dumps the contents of the body to screen.